src/components/ItemCard/ItemCard.jsx

```diff
@@ -0,0 +1,89 @@
+import { useState, useEffect } from 'react';
+import { getItemList, itemList } from '../../features/slices/item';
+import { useSelector, useDispatch } from '../../app/store';
+
+function ItemCard() {
+  const itemData = useSelector(itemList);
+  const [items, setItems] = useState([]);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getItemList(itemData));
+    setItems(itemData);
+  }, [items]);
```

Comment:
If you set the item as a dependency, then when the component first rendering will not show the items; I guess you need to replace it with **itemData** instead.

Reply:
it creates an infinite loop for some reason. i saw the error u're mentioning, trying to figure it out